tests/rptest/remote_scripts/compute_storage.py

```diff
@@ -109,7 +124,12 @@ def compute_size_for_file(file: Path, calc_md5: bool):
 
                 f.seek(0)
                 data = f.read()
-                reader = SegmentReader(io.BytesIO(data))
+
```

Comment:
Just curious, is there any particular reason to do the retries in read_batch vs wrapping the `with file.open('rb')` block in a retry in case of empty reads? Is it to avoid depending on some intermediary Exception type to signal the issue?

Reply:
Mainly because we already have the stream open within the segment reader, we just have to rewind back to the position before the batch in case of retry. 

If we retry at a higher level we would have to open the file again and then seek to the correct position, so the segment reader would need another argument (the offset to seek to).